seasmon_xr/ops/zonal.py

```diff
@@ -0,0 +1,43 @@
+"""Numba accelerated zonal statistics."""
+from numba import njit
+from numpy import zeros
+
+from ._helper import lazycompile
+
+
+@lazycompile(njit)
+def do_mean(pixels, z_pixels, nodata, num_zones, dtype="float64"):
+    """Calculate the zonal mean.
+
+    The mean for each pixel of `pixels` is calculated for each zone in
+    `z_pixels`.
+
+    The zones in `z_pixels` have to be numbered in a linear fashion,
+    starting with 0 for the first zone and num_zones for the last zone.
+
+    Args:
+        pixels: input value pixels (T,Y,X)
+        z_pixels: zonal pixels (Y,X)
+        nodata: nodata value in values
+        num_zones: number of zones in z_pixels
+        dtype: datatype
+    """
+    t, nr, nc = pixels.shape
+
+    sums = zeros((t, num_zones), dtype=dtype)
+    valids = zeros((t, num_zones), dtype=dtype)
+
+    for tix in range(t):
+        for rw in range(nr):
+            for cl in range(nc):
+                pix = pixels[tix, rw, cl]
+                z_idx = z_pixels[rw, cl]
+                if pix != nodata and z_idx >= 0:
+                    sums[tix, z_idx] += pix
+                    valids[tix, z_idx] += 1
+
+        for idx in range(sums.shape[1]):
+            if valids[tix, idx] > 0:
+                sums[tix, idx] = sums[tix, idx] / valids[tix, idx]
```

Comment:
I think output should be `nan` if there were no observations of a particular category at all, currently this will produce `0`.
```suggestion
                sums[tix, idx] = sums[tix, idx] / valids[tix, idx]
            else:
                sums[tix, idx] = float("nan")
            
```